src/engine/LocalVocab.cpp

```diff
@@ -46,7 +67,7 @@ LocalVocabIndex LocalVocab::getIndexAndAddIfNotContained(std::string&& word) {
 
 // _____________________________________________________________________________
 const std::string& LocalVocab::getWord(LocalVocabIndex localVocabIndex) const {
-  if (localVocabIndex.get() > idsToWordsMap_.size()) {
+  if (localVocabIndex.get() >= idsToWordsMap_.size()) {
```

Comment:
maybe use `at()` below instead of `operator[]` and let the compiler remove the redundant check (It should be that smart).